typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -39,12 +39,23 @@ export class MultiProvider {
     chainMetadata: ChainMap<ChainMetadata> = defaultChainMetadata,
     options: MultiProviderOptions = {},
   ) {
-    this.metadata = chainMetadata;
+    for (const chain of Object.values(chainMetadata)) {
+      this.addChain(chain);
+    }
+    this.logger = debug(options?.loggerName || 'hyperlane:MultiProvider');
+  }
+
+  addChain(metadata: ChainMetadata): void {
+    if (this.getKnownChainNames().includes(metadata.name)) {
+      throw new Error(`Duplicate chain name: ${name}`);
+    }
+    this.metadata[metadata.name] = metadata;
+    // TODO: This is O(n^2)
```

Comment:
Yeah but when n < 100 it's irrelevant right? The mem requirements of keeping a map make it a wash.

Reply:
Yeah for sure I don't think this matters, just wanted to call it out